src/middleware/parse-file/test/parse-file_test.js

```diff
@@ -0,0 +1,141 @@
+import assert from 'assert';
+import sinon from 'sinon';
+import sinonStubPromise from 'sinon-stub-promise';
+import EventEmitter from 'events';
+import proxyquire from 'proxyquire';
+import fs from 'fs';
+
+import index from '../index';
+import stream from 'stream';
+
+sinonStubPromise(sinon); //synchronous
+
+class MyEmitter extends EventEmitter {} //for mocking anything that emits events i.e. can be called `on('eventName', ...)` on
+
+const middleware = proxyquire('../', {
+  //overide import
+  'busboy': MyEmitter
+});
+
+const expectedParsed = [ 
+  { //_id: '589362430afa40f203d93cf8',
+    name: 1,
+    age: 19,
+    status: 'P',
+    likes: [ 'golf', 'racquetball' ] },
+  { //_id: '5893626b0afa40f203d93cf9',
+    name: 2,
+    age: 32,
+    status: 'P',
+    likes: [ 'golf', 'racquetball' ] 
+} ];
+
+export function testFileBeingPutOnRequest(done) {
+  const mockRes = {};
+  const mockReq = {
+    headers: {},
+    pipe: function(busboy) {
+      busboy.emit('file', 'test', fs.createReadStream('./fixture/import.json'), 'import.json', '7bit', 'application/json');
+    }
+  };
+  const underTest = middleware.default();
+
+  underTest(
+    mockReq, mockRes, err => {
+    assert.ok(mockReq.file);
+    assert.ok(!err);
+    done();
+  });
+}
+
+export function testParserExtendedJson(done) {
+  const mockRes = {};
+  const mockReq = {
+    headers: {},
+    pipe: function(busboy) {
+      busboy.emit('file', 'test', fs.createReadStream(__dirname + '/fixture/import.json'), 'import.json', '7bit', 'application/json');
+    }
+  };
+  const parsed = [];
+  const underTest = middleware.default();
+
+  underTest(
+    mockReq, mockRes, err => {
+      mockReq.file
+        .pipe(new BatchStream())
+        .on('finish', () => {
+          console.log('finish');
+          done();
+        });
+      assert.ok(!err);
+    });
+}
+
+export function testParserCsv(done) {
```

Comment:
Since these tests will be doing the same test can we do something like:

```javascript
const parserInfo = [
  {ext: 'csv', mimeType: 'text/csv'},
  {ext: 'json', mimeType: 'application/json'},
  {ext: 'bson', mimeType: 'application/octet-stream'}
];

export function testParsers(done) {
  parserInfo.forEach(parser => {
    const mockRes = {};
    const mockReq = {
      headers: {},
      pipe: function(busboy) {
        busboy.emit('file', 'test', fs.createReadStream('${__dirname}/fixture/import.${parser.ext}'), 'import.${parser.ext}', '7bit', parser.mimeType);
      }
    };

    const middleWare = parseFile.default();

    middleWare(mockReq, mockRes, err => {
      assert.ok(!err);

      mockReq.file.on('data', (obj) => {
        assert.deepEqual(expectedParsed.pop(), obj);
      }).on('finish', () => {
        done();
      });

    });
  })
}
```